src/js/components/Carousel/Carousel.js

```diff
@@ -1,242 +1,222 @@
+          setContainerProps({
+            heightProp: `${childRef.offsetHeight}px`,
+            widthProp: `${childRef.offsetWidth}px`,
+          });
+        } else setContainerProps({});
+      }
+    } else {
+      setContainerProps({
+        heightProp: fill ? '100%' : height,
+        widthProp: fill ? '100%' : width,
       });
     }
-  }
+  }, [noContainer, fill, height, width]);
 
-  const onChildChange = useCallback(
-    (index) => {
-      if (onChild) {
-        onChild(index);
-      }
-    },
-    [onChild],
-  );
+  /**
+   * Delays the transitions between Carousel slides. This is needed to
+   * avoid users "spamming" the controls which results in jarring animations
+   * and a bad user experience.
+   */
+  useEffect(() => {
+    let transitionTimer;
+    if (inTransition) {
+      transitionTimer = setTimeout(() => {
+        setInTransition(false);
+      }, 500);
+    }
+    return () => clearTimeout(transitionTimer);
+  }, [inTransition, setInTransition]);
 
+  // Handles auto-playing Carousel slides
   useEffect(() => {
+    let playTimer;
     if (play) {
-      const timer = setInterval(() => {
-        if (activeIndex < lastIndex) {
-          setIndexes({
-            activeIndex: activeIndex + 1,
-            priorActiveIndex: activeIndex,
-          });
-          onChildChange(activeIndex + 1);
-        } else {
-          setIndexes({
-            activeIndex: 0,
-            priorActiveIndex: activeIndex,
-          });
-          onChildChange(0);
-        }
+      playTimer = setInterval(() => {
+        onNext(current, inTransition);
       }, play);
-
-      timerRef.current = timer;
-
-      return () => {
-        clearTimeout(timer);
-      };
     }
-    return () => {};
-  }, [activeIndex, play, children, lastIndex, onChildChange]);
+    return () => clearTimeout(playTimer);
+  }, [current, inTransition, play, onNext]);
 
-  const onRight = () => {
-    if (activeIndex >= lastIndex) {
-      return;
-    }
-    clearInterval(timerRef.current);
-    setIndexes({
-      activeIndex: activeIndex + 1,
-      priorActiveIndex: activeIndex,
-    });
-    onChildChange(activeIndex + 1);
-  };
-
-  const onLeft = () => {
-    if (activeIndex <= 0) {
-      return;
-    }
-    clearInterval(timerRef.current);
-    setIndexes({
-      activeIndex: activeIndex - 1,
-      priorActiveIndex: activeIndex,
-    });
-    onChildChange(activeIndex - 1);
-  };
-
-  const onSelect = (index) => () => {
-    if (activeIndex !== index) {
-      clearInterval(timerRef.current);
-      setIndexes({ activeIndex: index, priorActiveIndex: activeIndex });
-      onChildChange(index);
+  // Allows Carousel slides to be controlled outside the component
+  useEffect(() => {
+    if (activeChild !== undefined && activeChild !== current) {
+      onJumpNavigation(current, activeChild, false);
     }
-  };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [activeChild]);
```

Comment:
Are there downsides to including 'current' and 'onJump...' as dependencies here?

Reply:
Yeah we only want to subscribe for `activeChild` updates. If current is modified and it runs those comparisons in the `useEffect` it will see that `activeChild !== current` and it will update the Carousel to the `activeChild` index even though it didn't get modified.

Reply:
We have gotten burned turning off exhaustive-deps in the past. It would be good to see if we can find a way to deliver the behavior we want without needing to disabled this.

Reply:
Rewriting this useEffect to adhere to lint rule 👍